code/modules/mob/living/basic/trader/trader.dm

```diff
@@ -0,0 +1,72 @@
+/mob/living/basic/trader
+	name = "Trader"
+	desc = "Come buy some!"
+	unique_name = FALSE
+	icon = 'icons/mob/simple/simple_human.dmi'
+	maxHealth = 200
+	health = 200
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	attack_verb_continuous = "punches"
+	attack_verb_simple = "punch"
+	attack_sound = 'sound/weapons/punch1.ogg'
+	basic_mob_flags = DEL_ON_DEATH
+	unsuitable_atmos_damage = 2.5
+	combat_mode = TRUE
+	move_resist = MOVE_FORCE_STRONG
+	mob_biotypes = MOB_ORGANIC|MOB_HUMANOID
+	sentience_type = SENTIENCE_HUMANOID
+	speed = 0
+
+	ai_controller = /datum/ai_controller/basic_controller/trader
+
+	///Sound used when item sold/bought
+	var/sell_sound = 'sound/effects/cashregister.ogg'
+	///The currency name
+	var/currency_name = "credits"
+	///The spawner we use to create our look
+	var/spawner_path = /obj/effect/mob_spawn/corpse/human/generic_assistant
+	///Our species to create our look
+	var/species_path = /datum/species/human
+	///The loot we drop when we die
+	var/loot = list(/obj/effect/mob_spawn/corpse/human/generic_assistant)
+	///Casing used to shoot during retaliation
+	var/ranged_attack_casing =/obj/item/ammo_casing/shotgun/buckshot
+	///Sound to make while doing a retalitory attack
+	var/ranged_attack_sound = 'sound/weapons/gun/pistol/shot.ogg'
+	///Weapon path, for visuals
+	var/held_weapon_visual = /obj/item/gun/ballistic/shotgun
+
+	///Type path for the trader datum to use for retrieving the traders wares, speech, etc
+	var/trader_data_path = /datum/trader_data
+
+/mob/living/basic/trader/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/trader, trader_data_path)
+	apply_dynamic_human_appearance(src, species_path = species_path, mob_spawn_path = spawner_path, r_hand = held_weapon_visual)
```

Comment:
put this proc first, then the components, then the elements. feels cleaner instead of comingling everything (unless we need teh trader component for the dynamic human appearance for some reason)